### PR TITLE
unit: add named derived units

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

I have not added L/dm^3 since I don't know how to handle that. It might just be necessary to us L rather than the correct unit.

Updates #730.

### CODE REVIEW

File: unit/generate_unit.go [hunk 1]
```diff
@@ -250,29 +251,267 @@ var Units = []Unit{
...
+		TypeComment: "Velocity represents a velocity in meters per second",
+		Dimensions: []Dimension{
+			{Name: LengthName, Power: 1},
+			{Name: TimeName, Power: -1},
+		},
+	},
+	{
+		Name:        "Voltage",
+		Receiver:    "v",
+		PrintString: "Pa",
```
Comment (btracey) [inline unit/generate_unit.go]: Volts, not pascals

File: unit/generate_unit.go [hunk 2]
```diff
@@ -250,29 +251,267 @@ var Units = []Unit{
...
+		Suffix:      "volt",
+		Singular:    "Volt",
+		TypeComment: "Voltage represents a voltage in Volts",
+		Dimensions: []Dimension{
+			{Name: CurrentName, Power: -1},
+			{Name: LengthName, Power: 2},
+			{Name: MassName, Power: 1},
+			{Name: TimeName, Power: -2},
```
Comment (btracey) [inline unit/generate_unit.go]: This should be -3, right?

File: unit/generate_unit.go [hunk 3]
```diff
@@ -250,29 +251,267 @@ var Units = []Unit{
...
+	},
+	{
+		Name:        "Inductance",
+		Receiver:    "i",
+		PrintString: "H",
+		Suffix:      "henry",
+		Singular:    "Henry",
+		TypeComment: "Inductance represents an electrical inductance in Henry",
+		Dimensions: []Dimension{
+			{Name: CurrentName, Power: -1},
```
Comment (btracey) [inline unit/generate_unit.go]: This should be -2?

File: unit/generate_unit.go (hunk 1 above)
Comment (kortschak) [reply unit/generate_unit.go]: I'm inventing a new system of unified units.

Comment (kortschak) [reply unit/generate_unit.go]: ... but fixed.

Comment (btracey) [reply unit/generate_unit.go]: Go is about making things simpler, afterall. I would probably have tried to start with Candela, but we can always get rid of that later.

File: unit/generate_unit.go (hunk 2 above)
Comment (kortschak) [reply unit/generate_unit.go]: Fixed

File: unit/generate_unit.go (hunk 3 above)
Comment (kortschak) [reply unit/generate_unit.go]: Fixed

**Comment by kortschak** [comment]
PTAL

**Comment by kortschak** [comment]
Queries:

1. Are you happy with the capitalisation of the units? There are many an varied ways to do this.
2. What should be do with volume? Just use m^3 or use L as the rendering and have liter-based constants with m^3 the underlying unit (I blame the chemists for this).

**Comment by btracey** [comment]
Assuming you're referring to things like "EquivalentRadioactiveDose", then I think the current choices are reasonable.

I think provide Volume in m^3, and then define L relative to that. I see cL and mL frequently, but I've never heard someone talk about megameters^3. It's sad that it's messy, but at least it's not imperial. There are aspects of imperial I really like, but volume is a total disaster.

**Comment by kortschak** [comment]
> Assuming you're referring to things like "EquivalentRadioactiveDose", then I think the current choices are reasonable.

No, I was thinking about initial capitals of Watt etc in the docs (some people write watt other Watt). Also pluralisation of units in docs. I have also made the Kelvin suffix "Kelvin" rather than "kelvin" (I don't know why  I should be consistent). Maybe we sort out this in the #90 handling of prefix constants?

**Comment by kortschak** [comment]
Sorry. PTAL

**Comment by btracey** [comment]
Fully sorting it out with 90 seems reasonable. I prefer megawatt to megaWatt, so I like the current way.

**Comment by kortschak** [comment]
Thanks for the review.

**Pull request merged**